engine/execution/rpc/engine.go

```diff
@@ -26,6 +28,16 @@ import (
 	"github.com/onflow/flow-go/utils/grpcutils"
 )
 
+var (
+	DefaultAPIRatelimits  map[string]int
+	DefaultAPIBurstlimits map[string]int
+)
+
+func init() {
+	DefaultAPIRatelimits = make(map[string]int)
+	DefaultAPIBurstlimits = make(map[string]int)
+}
+
```

Comment:
Not sure what the purpose of these is. It seem they are literally initiating empty `map`s, which are used as the default values when loading the flags, right? then why not initiating and passing them at the `execution_builder`? e.g., 

```
flags.StringToIntVar(&e.exeConf.apiRatelimits, "api-rate-limits", make(map[string]int), "per second rate limits for GRPC API methods e.g. Ping=300,ExecuteScriptAtBlockID=500 etc.")
```

Reply:
The empty maps were just placeholders while I determine the values to use for each endpoint. That is done in this commit https://github.com/onflow/flow-go/pull/2505/commits/91fa062ad2832ee5d958cec35cb333a32b424df8 . I took a look at the req/s handled by each endpoint and cross-referenced it with the consensus rate and execution rate over the past 6 months. I did not find any correlation between spikes in req/s and consensus/block rate. Thus I took the peak req/s value for each endpoint over the past 6 months and used double that value as the req/s config value for rate limiting.

Reply:
We have decided to not use any default values for rate limiting, and that it should be optional for the node operator. https://github.com/onflow/flow-go/pull/2505/commits/792e3338e29debfba777e410d61f272659389be4